contracts/XtraBBroRewarder.sol

```diff
@@ -81,39 +73,57 @@ contract BBroPremiumStakingRewardsDistributor is Ownable {
         }
     }
 
-    function _getAmountForReward(address _staker)
+    function _calculateXtraBBroReward(address _staker)
         private
         view
         returns (uint256)
     {
-        uint256 amountForReward = 0;
-
         IStakingV1.Staker memory staker = staking.getStakerInfo(_staker);
+
+        uint256 bbroXtraReward = 0;
         for (uint256 i = 0; i < staker.unstakingPeriods.length; i++) {
             if (
                 staker.unstakingPeriods[i].unstakingPeriod >=
                 minUnstakingPeriodForXtraReward
             ) {
-                amountForReward +=
+                bbroXtraReward += _computeBBroReward(
                     staker.unstakingPeriods[i].rewardsGeneratingAmount +
-                    staker.unstakingPeriods[i].lockedAmount;
+                        staker.unstakingPeriods[i].lockedAmount,
+                    staker.unstakingPeriods[i].unstakingPeriod
+                );
             }
         }
 
-        return amountForReward;
+        for (uint256 i = 0; i < staker.withdrawals.length; i++) {
+            if (
+                staker.withdrawals[i].unstakingPeriod >=
+                minUnstakingPeriodForXtraReward
+            ) {
+                bbroXtraReward += _computeBBroReward(
+                    staker.withdrawals[i].rewardsGeneratingAmount +
+                        staker.withdrawals[i].lockedAmount,
+                    staker.withdrawals[i].unstakingPeriod
+                );
+            }
+        }
+
+        if (terraMigratorsWhitelist[_staker]) {
+            bbroXtraReward = (bbroXtraReward * terraMigratorExtraPerc) / 100;
```

Comment:
Please double check with @shroshko if this formula does what we expect to do.

It feels to me that if we use a terraMirgratorExtraPerc of 1.6, we increase the rewards to much for them because first we increase it by 1.4x, and then again by 1.6x